src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -255,20 +262,26 @@ def build_vm_resource(  # pylint: disable=too-many-locals, too-many-statements
         computer_name=None, dedicated_host=None, priority=None, max_price=None, eviction_policy=None,
         enable_agent=None, vmss=None, os_disk_encryption_set=None, data_disk_encryption_sets=None, specialized=None,
         encryption_at_host=None, dedicated_host_group=None, enable_auto_update=None, patch_mode=None,
-        enable_hotpatching=None):
+        enable_hotpatching=None, count=None):
 
     os_caching = disk_info['os'].get('caching')
 
     def _build_os_profile():
 
         special_chars = '`~!@#$%^&*()=+_[]{}\\|;:\'\",<>/?'
 
+        # 2 is used to avoid shadow names
+        computer_name2 = computer_name or ''.join(filter(lambda x: x not in special_chars, name))
```

Comment:
how about name it better such as _computer_name

Reply:
fixed